Lesson16/Emirhan/script.js

```diff
@@ -1,97 +1,77 @@
 /*
 8. Print Each Character of a String (For-of)
- - Define a function `printCharacters(str)` that uses a for-of loop
-   to log each character in the string on a separate line.
+   - Define a function `printCharacters(str)` that uses a for-of loop
+     to log each character in the string on a separate line.
 */
-
 function printCharacters(str) {
-  for (const char of str) {
-    console.log(char);
+  for (let character of str) {
+    console.log(character);
   }
 }
-
-printCharacters('Mila');
-
+printCharacters('function');
 /*
 9. Sum All Values in an Object
    - Define a function `sumObjectValues(obj)` that iterates over the
      properties of an object (using a for-in loop) and sums all numeric values.
    - Log the sum.
    - Example: {a: 10, b: 20, c: 5} -> 35
 */
-
-const obj = { a: 10, b: 20, c: 5 };
-
 function sumObjectValues(obj) {
-  let sum = 0;
-  for (const key in obj) {
-    if (typeof obj[key] === 'number') {
-      sum += obj[key];
-    }
+  let result = 0;
+  for (let values in obj) {
+    result += obj[values];
   }
-  return sum;
+  console.log(result);
 }
-
-console.log(sumObjectValues({ a: 10, b: 20, c: 5 }));
+sumObjectValues({ a: 10, b: 20, c: 5 });
 
 /*
 10. Print Keys of an Object (For-in)
     - Define a function `printObjectKeys(obj)` that uses a for-in loop
       to log each key of the object.
     - Example: { name: "Alice", age: 25 } -> logs "name", then "age"
 */
-const testObject = {
-  name: 'Mila',
-  age: 5,
-};
-
 function printObjectKeys(obj) {
-  for (const key in obj) {
-    console.log(key + ':' + obj[key]);
+  for (let key in obj) {
+    console.log(`${key} `);
   }
 }
-printObjectKeys(testObject);
+printObjectKeys({ name: 'Alice', age: 25 });
 /*
 11. Sum Array Using do-while Loop
     - Define a function `sumWithDoWhile(numbers)` that uses a do-while loop
       to sum all numbers in the array.
     - Log the total.
 */
-
-const numbers = [1, 2, 3, 4, 5];
-
 function sumWithDoWhile(numbers) {
-  let sum = 0;
+  result = 0;
```

Comment:
It's important to define a variable with let/const keyword, otherwise the variables is being defined in Global scope which may cause unexpected issues.
```suggestion
  let result = 0;
```